sdk/core/core-http-compat/src/extendedClient.ts

```diff
@@ -50,4 +56,41 @@ export class ExtendedServiceClient extends ServiceClient {
       });
     }
   }
+
+  /**
+   * Compatible send operation request function.
+   *
+   * @param operationArguments - Operation arguments
+   * @param operationSpec - Operation Spec
+   * @returns
+   */
+  async sendOperationRequest<T>(
+    operationArguments: OperationArguments,
+    operationSpec: OperationSpec
+  ): Promise<T> {
+    const userProvidedCallBack: RawResponseCallback | undefined =
+      operationArguments?.options?.onResponse;
+
+    if (!operationArguments.options) operationArguments.options = {};
+    let lastResponse: FullOperationResponse | undefined;
+
+    function onResponse(
+      rawResponse: FullOperationResponse,
+      flatResponse: unknown,
+      error?: unknown
+    ): void {
+      lastResponse = rawResponse;
+      if (userProvidedCallBack) {
+        userProvidedCallBack(rawResponse, flatResponse, error);
+      }
+    }
+
+    operationArguments.options.onResponse = onResponse;
+    const result: T = await super.sendOperationRequest(operationArguments, operationSpec);
+    Object.defineProperty(result, "_response", {
+      value: lastResponse,
```

Comment:
We also need to transform _response to be compatible with core-http's HttpResponse? Probably hard to achieve100% compatible though

Reply:
@jeremymeng I have compared the `_response` between `core-http` and `core-client`. The structure is

**core-http**
```js
export interface HttpResponse {
  headers: HttpHeadersLike;
  status: number;  
  request: WebResourceLike;
}

export interface HttpOperationResponse extends HttpResponse {
  parsedHeaders?: { [key: string]: any };
  parsedBody?: any;
  bodyAsText?: string | null;
  blobBody?: Promise<Blob>;
  readableStreamBody?: NodeJS.ReadableStream;
}
```

**core-client**
```js
export declare interface PipelineResponse {
  request: PipelineRequest;
  status: number;
  headers: HttpHeaders;
  bodyAsText?: string | null;
  readableStreamBody?: NodeJS.ReadableStream;
  blobBody?: Promise<Blob>;
  browserStreamBody?: ReadableStream<Uint8Array>;
}

export declare interface FullOperationResponse extends PipelineResponse {
  parsedHeaders?: { [key: string]: unknown; };
  parsedBody?: any;  
  request: OperationRequest;
}
```

As you can see, all the fields in `core-http` are available in `core-client` also. Do you want the types also to be matched? For example, do you suggest to build an adapter between `HttpHeaders` from `core-client` and `HttpHeadersLike` from `core-http`?

Reply:
> Do you want the types also to be matched?

Yes that's what I am thinking. I added something like this while working on service bus atom client (probably want to add `body` too but I removed things that I don't feel people would access): https://github.com/Azure/azure-sdk-for-js/blob/31c1a6ddf9f58dccd8f755004b9c228896efbd7c/sdk/servicebus/service-bus/src/util/compat/compatibility.ts#L125-L135

Reply:
@jeremymeng What is the need to convert this to WebResourceLike object? Are we exposing this and asking the customers to use it in a certain way? Or is it just to avoid the breaking change?

Reply:
Also, isn't this a request object within the response? How are our customers using this? Are there any use case scenarios available?

Reply:
Just try to avoid breaking change (in case any existing customers use any of those)

Reply:
@jeremymeng Thanks. I have used the code from service bus to here and modified the code and verified it. Please review.